public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.test.ts

```diff
@@ -210,3 +299,91 @@ describe('transformSaveModelSchemaV2ToScene', () => {
     expect(getQueryRunnerFor(vizPanels[0])?.state.datasource?.uid).toBe(MIXED_DATASOURCE_NAME);
   });
 });
+
+interface VariableValidation<T extends TypedVariableModelv2> {
+  variable: SceneVariable<SceneVariableState> | undefined;
+  kind: T;
+  scene: DashboardScene;
+  dashSpec: DashboardV2Spec;
+  index: number;
+}
+
+function validateQueryVariable({ variable, kind, scene, dashSpec, index }: VariableValidation<QueryVariableKind>) {
+  expect(variable).toBeInstanceOf(QueryVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
+
+function validateCustomVariable({ variable, kind, scene, dashSpec, index }: VariableValidation<CustomVariableKind>) {
+  expect(variable).toBeInstanceOf(CustomVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
+
+function validateDataSourceVariable({
+  variable,
+  kind,
+  scene,
+  dashSpec,
+  index,
+}: VariableValidation<DatasourceVariableKind>) {
+  expect(variable).toBeInstanceOf(DataSourceVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
+
+function validateConstantVariable({
+  variable,
+  kind,
+  scene,
+  dashSpec,
+  index,
+}: VariableValidation<ConstantVariableKind>) {
+  expect(variable).toBeInstanceOf(ConstantVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
+
+function validateIntervalVariable({
+  variable,
+  kind,
+  scene,
+  dashSpec,
+  index,
+}: VariableValidation<IntervalVariableKind>) {
+  expect(variable).toBeInstanceOf(IntervalVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
+
+function validateTextBoxVariable({ variable, kind, scene, dashSpec, index }: VariableValidation<TextVariableKind>) {
+  expect(variable).toBeInstanceOf(TextBoxVariable);
+  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
+    kind.spec.current.value
+  );
+}
```

Comment:
Suggestion to reduce those function

```ts
function validateVariable({ variable, kind, scene, dashSpec, index, sceneObjectClass }) {
  expect(variable).toBeInstanceOf(sceneObjectClass);
  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
    kind.spec.current.value
  );
} 


validateVariable({....., TextVariable});
```



```suggestion
function validateTextBoxVariable({ variable, kind, scene, dashSpec, index }: VariableValidation<TextVariableKind>) {
  expect(variable).toBeInstanceOf(TextBoxVariable);
  expect(scene.state?.$variables?.getByName(dashSpec.variables[index].spec.name)?.getValue()).toBe(
    kind.spec.current.value
  );
}
```